appinventor/blocklyeditor/src/blocks/utilities.js

```diff
@@ -75,6 +75,29 @@ Blockly.Blocks.Utilities.wrapSentence = function(str, len) {
   }
 };
 
+// Change the text of collapsed blocks on rename
+// Recurse to fix collapsed parents
+
+Blockly.Blocks.Utilities.MAX_COLLAPSE = 4;
```

Comment:
@jisqyv 

The recursive check is used when parents of "renamed blocks" are collapsed. For example, 

![capture2](https://cloud.githubusercontent.com/assets/8927635/9485892/8bc6fb88-4bdd-11e5-9201-61453248be24.PNG)

The "if" block is collapsed, but uses the "get name" in its collapsed string (`_TEMP_COLLAPSED_INPUT`).

Since `_TEMP_COLLAPSED_INPUT` already has a fixed string length (30, I think) - I couldn't find instances where more than 2 recursive parents would contain the "renamed block". 4 seemed to be a safe number of parents to check.